src/main/java/org/folio/gobi/Mapper.java

```diff
@@ -31,97 +30,121 @@ public Mapper(Map<Mapping.Field, DataSourceResolver> mappings) {
       mapEresource(futures, eresource, doc);
-      mapFundDistibution(futures, fundDistribution, doc);
-      mapLocation(futures, location, doc);
-      mapVendorDetail(futures, vendorDetail, doc);
-      mapClaims(futures, claim, doc);
-      mapRenewal(futures, renewal, doc);
+    } else {
       mapPhysical(futures, physical, doc);
-      mapSource(futures, source, doc);
-      mapContributor(futures, contributor, doc);
-      mapReportingCodes(futures, reportingCode, doc);
-      mapVendorDependentFields(futures, eresource, physical, compPO, claim, doc);
-      mapLicense(futures, license, doc);
-
-      CompletableFuture.allOf(futures.toArray(new CompletableFuture<?>[futures.size()]))
-        .thenAccept(v -> {
-          compPO.setTotalItems(location.getQuantity());
-
-          setObjectIfPresent(detail, o -> pol.setDetails((Details) o));
-          setObjectIfPresent(cost, o -> pol.setCost((Cost) o));
-          setObjectIfPresent(license, o -> eresource.setLicense((License) o));
-          if (pol.getOrderFormat() == CompositePoLine.OrderFormat.ELECTRONIC_RESOURCE) {
-            setObjectIfPresent(eresource, o -> pol.setEresource((Eresource) o));
-          }
-          setObjectIfPresent(vendorDetail, o -> pol.setVendorDetail((VendorDetail) o));
-          setObjectIfPresent(renewal, o -> compPO.setRenewal((Renewal) o));
-          if (pol.getOrderFormat() == CompositePoLine.OrderFormat.PHYSICAL_RESOURCE) {
-            setObjectIfPresent(physical, o -> pol.setPhysical((Physical) o));
-          }
-          setObjectIfPresent(source, o -> pol.setSource((Source) o));
-
-          setObjectIfPresent(location, o -> {
-            List<Location> locations = new ArrayList<>();
-            locations.add(location);
-            pol.setLocations(locations);
-          });
-          setObjectIfPresent(contributor, o -> {
-            List<Contributor> contributors = new ArrayList<>();
-            contributors.add(contributor);
-            pol.setContributors(contributors);
-          });
-          setObjectIfPresent(reportingCode, o -> {
-            List<ReportingCode> reportingCodes = new ArrayList<>();
-            reportingCodes.add(reportingCode);
-            pol.setReportingCodes(reportingCodes);
-          });
-          setObjectIfPresent(claim, o -> {
-            List<Claim> claims = new ArrayList<>();
-            claims.add(claim);
-            pol.setClaims(claims);
-          });
-          setObjectIfPresent(fundDistribution, o -> {
-            List<FundDistribution> fundDistributions = new ArrayList<>();
-            fundDistributions.add(fundDistribution);
-            pol.setFundDistribution(fundDistributions);
-          });
-          poLines.add(pol);
-
-          compPO.setCompositePoLines(poLines);
-          future.complete(compPO);
-        });
-    } catch (Exception e) {
-      throw new CompletionException(e);
     }
 
+    CompletableFuture.allOf(futures.toArray(new CompletableFuture<?>[futures.size()]))
```

Comment:
According to comment https://github.com/folio-org/mod-orders/pull/32#discussion_r239613204
it is good practice to not predefine array size, but to left it zero.

Reply:
Done